packages/controls/src/serialization/function-serialization.test.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  deserializeFunction,
+  serializeFunction,
+} from './function-serialization'
+
+describe('function serialization', () => {
+  test('correctly deserializes serialized function', async () => {
+    const getItemLabel = (item: { label: string }) => item.label
+    const serialized = serializeFunction(getItemLabel)
+
+    const deserialized1 = deserializeFunction(serialized)
+    const deserialized2 = deserializeFunction(serialized)
```

Comment:
Deserializing the same serialized function twice used to "orphan" the previously deserialized function instances -- see below.